cmd/satellite/reports/attribution.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package reports
+
+import (
+	"context"
+	"encoding/csv"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/skyrings/skyring-common/tools/uuid"
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+
+	"storj.io/storj/internal/memory"
+	"storj.io/storj/satellite/attribution"
+	"storj.io/storj/satellite/satellitedb"
+)
+
+// GenerateValueAttributionCSV creates a report with
+func GenerateValueAttributionCSV(ctx context.Context, database string, partnerID uuid.UUID, start time.Time, end time.Time, output io.Writer) error {
+	db, err := satellitedb.New(zap.L().Named("db"), database)
+	if err != nil {
+		return errs.New("error connecting to master database on satellite: %+v", err)
+	}
+	defer func() {
+		err = errs.Combine(err, db.Close())
+	}()
+
+	rows, err := db.Attribution().QueryValueAttribution(ctx, partnerID, start, end)
+	if err != nil {
+		return err
+	}
+
+	w := csv.NewWriter(output)
+	headers := []string{
+		"projectID",
+		"bucketName",
+		"byte-hours:Remote",
+		"byte-hours:Inline",
+		"bytes:BWEgress",
+	}
+	if err := w.Write(headers); err != nil {
+		return err
+	}
+
+	for _, row := range rows {
+		record, err := csvRowToStringSlice(row)
+		if err != nil {
+			return err
+		}
+		if err := w.Write(record); err != nil {
+			return err
+		}
+	}
+	if err := w.Error(); err != nil {
+		return err
+	}
+	w.Flush()
+	if output != os.Stdout {
```

Comment:
Why message only for `Stdout`?

Reply:
Sorry, why only for file?

Reply:
I followed the pattern of the storagenode-usage report.  I guess they thought it was only necessary to provide the message if the data was going to a file.

Reply:
hmm I would guess that should be opposite but if it works for payments then ok.